plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+        self.log(str(require_update), "INFO")
+        if not require_update:
+            msg = "The update for AP Config '{0}' has been successfully verified.".format(ap_name)
+            self.log(msg, "INFO")
+            self.status = "success"
+            responses = {}
+            responses["accesspoints_verify"] = {"want": self.want,
+                                                "have": self.payload["device_config"],
+                                                "message": msg}
+            self.result['response'].append(responses)
+            return self
+
+        self.log("""The playbook input for AP '{0}' does not align with the Cisco Catalyst Center, indicating that the merge task
+                 may not have executed successfully.""".format(ap_name), "INFO")
+
+        return self
+
+    def get_current_config(self, input_config):
+        """
+        Check if the input wireless accesspoint configuration exists in Cisco Catalyst Center.
+
+        Parameters:
+          - self (object): An instance of the class containing the method.
+        Returns:
+            A Dictionary list contains Accesspoint configuration based on the input given from
+            playbook either mac_address or management_ip_address or hostname
+            [
+                {
+                    "ap_name": "NFW-AP1-9130AXE",
+                    "eth_mac": "34:5d:a8:0e:20:b4",
+                    "led_brightnessLevel": 3,
+                    "led_status": "Enabled",
+                    "location": "LTTS",
+                    "mac_address": "90:e9:5e:03:f3:40"
+                }
+            ]
+        Description:
+            Checks the existence of a Device details in Cisco Catalyst Center by querying the
+            'get_device_list' function in the 'devices' family. Based on the response need to 
+            pass the ap_ethernet_mac_address to the another 'get_access_point_configuration' function
+            in the 'wireless' family to get AP configuration to check the input data with current
+            config data and return above resoponse.
+        """
+        accesspoint_exists = False
+        current_configuration = {}
+        response = None
+        input_param = {}
+        if input_config.get("mac_address") is not None and input_config.get("mac_address") != "":
+            input_param["macAddress"] = input_config["mac_address"]
+        elif input_config.get("management_ip_address") is not None and \
+            input_config.get("management_ip_address") != "":
+            input_param["managementIpAddress"] = input_config["management_ip_address"]
+        elif input_config.get("hostname") is not None and input_config.get("hostname") != "":
+            input_param["hostname"] = input_config["hostname"]
+
+        if not input_param:
+            self.log("Required param mac_address, management_ip_address or hostname is not in playbook config",
```

Comment:
If the required parameter is not present, then it's a failure. We should failed from here itself and stop the execution.

Reply:
for update we need these param but creating AP config. this is not mandatory to identify to create or update this if condition required. unless need to route to create AP config.